Added formatter to format full lat/long string

tstibbs opened the pull request:
This pull request adds the ability to specify a formatter that formats the lat/long as a whole.

This can be useful when using unusual/custom coordinate systems and projections.

NB: the diff makes it look like many lines have changed but most of them are just indentation changes.

xguaita commented:
That is a good idea

If you add only this code in _getLatLngCoord (line 91), avoids the extra indentation:
`if (this.options.latLngFormatter !== undefined) return this.options.latLngFormatter(center.lat, center.lng);`

Please make the changes and I will merge with master

It would be great to add an example

tstibbs committed:
Added formatter to format full lat/long string

tstibbs commented:
Fair enough, I've made that change.

I've also updated the readme to show how to use the function, and added an new example file that shows it in action (`examples/latlng_formatter.html`). The only example I can think of that this would be used for is something like the UK OSGB alphanumeric grid references, and these require quite a bit of code to calculate, so in the example I've given a simpler string concatenation that just shows how it gets wired up.

tstibbs committed:
Better lat/long formatter example
